http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -180,15 +216,47 @@ protected String getRedirectUri(String state) {
             if (deployment.getAuthUrl() == null) {
                 return null;
             }
-            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl())
-                    .addParameter(RESPONSE_TYPE, CODE)
-                    .addParameter(CLIENT_ID, deployment.getResourceName())
-                    .addParameter(REDIRECT_URI, rewrittenRedirectUri(url))
-                    .addParameter(STATE, state);
-            redirectUriBuilder.addParameters(forwardedQueryParams);
+
+            String redirectUri = rewrittenRedirectUri(url);
+            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl());
+            redirectUriBuilder.addParameter(RESPONSE_TYPE, CODE)
+                    .addParameter(CLIENT_ID, deployment.getResourceName());
+            if (deployment.getAuthenticationRequestFormat().contains(REQUEST_TYPE_OAUTH2)) {
+                redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri)
+                        .addParameter(STATE, state)
+                        .addParameters(forwardedQueryParams);
+            } else if (deployment.getAuthenticationRequestFormat().contains(REQUEST_TYPE_REQUEST_URI)) {
+                if (deployment.getRequestUriParameterSupported()) {
+                    String request = convertToRequestParameter(redirectUriBuilder, redirectUri, state, forwardedQueryParams);
+                    String request_uri = getRequestUri(request);
+                    redirectUriBuilder.addParameter(REQUEST_URI, request_uri);
+                    redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri);
+                } else {
+                    // send request as usual
+                    redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri)
```

Comment:
Since there are three cases where the request is sent in the usual way, it would be good to introduce a helper method for this so this code for updating the redirectUriBuilder doesn't need to be repeated in three places.

Reply:
Helper methods could also be introduced to set the values for the other cases as well to simplify the different cases here and make it a bit easier to read.

Reply:
Done. 
Please feel free to let me know if there is anything I should change.